operator/v1/types_dns.go

```diff
@@ -130,6 +131,61 @@ type Server struct {
 	ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
 }
 
+// DNSTransport indicates what type of connection should be used
+// +kubebuilder:validation:Enum=tls;cleartext
+type DNSTransport string
+
+const (
+	// TLSTransport indicates that TLS should be used for the connection
+	TLSTransport DNSTransport = "tls"
+
+	// CleartextTransport indicates that no encryption should be used for
+	// the connection.
+	CleartextTransport DNSTransport = "cleartext"
+)
+
+// DNSOverTLSConfig groups related configuration parameters used for configuring
+// forwarding to upstream resolvers that support DNS-over-TLS
+//
+type DNSOverTLSConfig struct {
+	// transport allows cluster administrators to opt-in to using a DNS-over-TLS
+	// connection between cluster DNS and an upstream resolver(s). Configuring
+	// TLS as the transport at this level without configuring a CABundle will
+	// result in the system certificates being used to verify the serving
+	// certificate of the upstream resolver(s).
+	//
+	// Possible values:
+	// "" (empty) - This is the default and results in DNS queries being sent over cleartext.
+	// "cleartext" - Cluster admin specified cleartext option. This results in the same functionality
+	// as an empty value but may be useful when a cluster admin wants to be more explicit about the transport,
+	// or wants to switch from "tls" to "cleartext" explicitly.
+	// "tls" - This indicates that DNS queries should be sent over a TLS connection. If Transport is set to TLS,
+	// you MUST also set ServerName.
+	//
+	// +optional
+	Transport DNSTransport `json:"transport,omitempty"`
+
+	// serverName is the upstream server to connect to for DNS resolution.
+	// This is required when Transport is set to "tls" and if Transport is empty or "cleartext"
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9\p{S}\p{L}]((-?[a-zA-Z0-9\p{S}\p{L}]{0,62})?)|([a-zA-Z0-9\p{S}\p{L}](([a-zA-Z0-9-\p{S}\p{L}]{0,61}[a-zA-Z0-9\p{S}\p{L}])?)(\.)){1,}([a-zA-Z0-9-\p{L}]){2,63})$`
+	ServerName string `json:"serverName,omitempty"`
```

Comment:
structured in a `DNSTransportConfig` struct, this should be done like a discriminated union with CABundle and ServerName combine in a  

```go
TLS *DNSOverTLSConfig

type DNSOverTLSConfig struct {
  ServerName string
  CABundle configv1.ConfigMapNameReference
}
```